languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -17643,14 +17643,29 @@ though, and has since been slightly modified)-->
             </rule>
         </rulegroup>
         <rulegroup id="NON_V" name="pronom sujet + nom/adj">
+        <antipattern>
+        <token>sur</token>
+        <token>-</token>
+        <token postag="V.*" postag_regexp="yes"/>
+        </antipattern>
+            <rule>
             <antipattern>
-                <token>Pearl</token>
-                <token regexp="yes">Harbou?r</token>
+            <token postag="R pers suj.*" postag_regexp="yes"/>
+            <token postag="(?!PARA_END)[ACKNJZP].*" postag_regexp="yes" regexp="yes">(?-i)[A-Z].*</token>
+            </antipattern>
+            <antipattern>
+            <token postag="R pers suj.*" postag_regexp="yes"/>
+            <token postag="C coor"/>
+            <token postag="R pers suj.*" postag_regexp="yes"/>
+            </antipattern>
+            <antipattern>
+                <token postag="R pers obj 2 s"/>
+                <token postag="R pers suj.*" postag_regexp="yes"/>
+                <token postag="(?!PARA_END)[ACKNJZP].*" postag_regexp="yes"/>
             </antipattern>
-            <rule>
                 <pattern>
-                    <token regexp="yes">je|tu|ils?|on
-                        <exception scope="previous">-</exception></token>
+                        <token postag="R pers suj.*" postag_regexp="yes">
+                            <exception scope="previous">-</exception></token>
```

Comment:
I am afraid this is too general. There will be FPs. Anyway this rule is being rewritten.